### PR TITLE
Upgrade API spec version to 2023-06-01

### PR DESCRIPTION
**Pull request by dubiety**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/maps/Azure.ResourceManager.Maps/src/Generated/Models/Encryption.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+
+namespace Azure.ResourceManager.Maps.Models
+{
+    /// <summary> (Optional) Discouraged to include in resource definition. Only needed where it is possible to disable platform (AKA infrastructure) encryption. Azure SQL TDE is an example of this. Values are enabled and disabled. </summary>
+    public partial class Encryption
```
Comment (archerzz) [inline sdk/maps/Azure.ResourceManager.Maps/src/Generated/Models/Encryption.cs:14]: The name is too generic. Consider appending RP prefix or renaming it.
- use `prepend-rp-prefix` to automatically append PR name. See example [here](https://github.com/Azure/azure-sdk-for-net/blob/176fd476588e8e61ebce747d66714cb39947cae5/sdk/azurestackhci/Azure.ResourceManager.Hci/src/autorest.md?plain=1#L52-L57).
- use `rename-mapping` to rename a model. See example [here](https://github.com/Azure/azure-sdk-for-net/blob/176fd476588e8e61ebce747d66714cb39947cae5/sdk/azurestackhci/Azure.ResourceManager.Hci/src/autorest.md?plain=1#L58-L76)

File: sdk/maps/Azure.ResourceManager.Maps/src/Generated/Models/IdentityType.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+#nullable disable
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.ResourceManager.Maps.Models
+{
+    /// <summary> Values can be systemAssignedIdentity or userAssignedIdentity. </summary>
+    public readonly partial struct IdentityType : IEquatable<IdentityType>
```
Comment (archerzz) [inline sdk/maps/Azure.ResourceManager.Maps/src/Generated/Models/IdentityType.cs:14]: name is too generic

File: sdk/maps/Azure.ResourceManager.Maps/src/Generated/Models/InfrastructureEncryption.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+#nullable disable
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.ResourceManager.Maps.Models
+{
+    /// <summary> Values are enabled and disabled. </summary>
+    public readonly partial struct InfrastructureEncryption : IEquatable<InfrastructureEncryption>
```
Comment (archerzz) [inline sdk/maps/Azure.ResourceManager.Maps/src/Generated/Models/InfrastructureEncryption.cs:14]: name is too generic

**Review by archerzz — CHANGES_REQUESTED** [review]
Generated code is not up to date.
    You may need to rebase on the latest main, 
    run 'eng\scripts\Update-Snippets.ps1' if you modified sample snippets or other *.md files (https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#updating-sample-snippets), 
    run 'eng\scripts\Export-API.ps1' if you changed public APIs (https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#public-api-additions). 
    run 'dotnet build /t:GenerateCode' to update the generated code and samples.
    
To reproduce this error locally, run 'eng\scripts\CodeChecks.ps1 -ServiceDirectory maps'.

**Comment by dubiety** [comment]
Hi @archerzz , due to some technical issue, I've created another PR (#47239) and address your comments.

I've update the naming for `Encryption`, `InfrastructureEncryption`, and `IdentityType` with prepend RP name. Please take a look at the new PR. Thank you very much!

**Pull request closed without merging**